f3/granite.go

```diff
@@ -460,6 +477,25 @@ func (i *instance) tryCommit(round uint32) {
 	}
 }
 
+func (i *instance) tryDecide(value ECChain) {
+	weakQuorum := i.decision.HasWeakQuorumAgreement(value.HeadCIDOrZero())
+	strongQuorum := i.decision.HasStrongQuorumAgreement(value.HeadCIDOrZero())
+
+	// If a weak quorum of DECIDE messages has been received, at least one correct participant must have sent one
+	// (and there exists a proof of it, albeit not necessarily locally present).
+	// That means that this value must be decided by every correct participant.
+	// Broadcasting a DECIDE message is necessary for liveness if the broadcast primitive is only best-effort broadcast.
+	// If broadcast() implemented reliable broadcast, this step would not be necessary.
+	if weakQuorum && !i.decideSent {
+		i.broadcast(DECIDE, value, nil, 0)
+		i.decideSent = true
+	}
+
+	if strongQuorum {
+		i.decide(value, i.round)
```

Comment:
We might want to rename this method  since "deciding" (entering the DECIDE phase) is no longer the terminating step, which is what this method is about. Perhaps `complete()` or `terminate()`?

Reply:
done (`terminate()`)